enos/modules/k8s_vault_verify_build_date/main.tf

```diff
@@ -0,0 +1,64 @@
+
+terraform {
+  required_providers {
+    enos = {
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+  }
+}
+
+locals {
+  // build date was introduced in Vault 1.11.0
+  requires_build_date_check = length(regexall("^(?:1\\.1[^0]|[2-9]|\\d{2,})", var.vault_product_version)) > 0
```

Comment:
So sorry- I see that my change actually missed a case.
Example of the _new_, working expression: [long link](https://pythex.org/?regex=%5E(%3F%3A%5B2-9%5D)%7C%5E(%3F%3A1%5C.%5B1-9%5D%5B1-9%5D%2B)%7C%5E1%5Cd&test_string=0.0.1%0A1.1.0%0A1.10.0%0A1.10.12%0A1.11.0%0A1.11.1%0A1.11.12%0A2.10.0%0A2.11.0%0A3.1.2%0A4.458.80%0A14.0.1%0A54.123.0%0A12341234.000000000.123412334812349rcwhatever&ignorecase=0&multiline=1&dotall=0&verbose=0)

Example of the previous expression's failure: [long link](https://pythex.org/?regex=%5E(%3F%3A1%5C.1%5B%5E0%5D%7C%5B2-9%5D%7C%5Cd%7B2%2C%7D)&test_string=0.0.1%0A1.1.0%0A1.10.0%0A1.10.12%0A1.11.0%0A1.11.1%0A1.11.12%0A2.11.0%0A3.1.2%0A4.458.80%0A14.0.1%0A54.123.0%0A12341234.000000000.123412334812349rcwhatever&ignorecase=0&multiline=1&dotall=0&verbose=0)

```suggestion
  requires_build_date_check = length(regexall("^(?:[2-9])|^(?:1\\.[1-9][1-9]+)|^1\\d", var.vault_product_version)) > 0
```

Verification that this regex is valid in Terraform:
```
> regexall("^(?:[2-9])|^(?:1\\.[1-9][1-9]+)|^1\\d", "0.0.0")
tolist([])
> regexall("^(?:[2-9])|^(?:1\\.[1-9][1-9]+)|^1\\d", "1.10.0")
tolist([])
> regexall("^(?:[2-9])|^(?:1\\.[1-9][1-9]+)|^1\\d", "1.1.0")
tolist([])
> regexall("^(?:[2-9])|^(?:1\\.[1-9][1-9]+)|^1\\d", "14.10.0")
tolist([
  "14",
])
```

Reply:
Might be better to use Enos `semver` function from the scenario step and pass this as a step variable instead ?

Reply:
@jaymalasinha Yes, I'm going to go with that. I just had the same revelation

Reply:
I did the re-req review for myself based on this information so that I was no longer requesting changes.

Reply:
Removed this altogether and replaced it with a semverconstraint function all in the scenario file, good by regex.